webviz_subsurface/plugins/_group_tree/_ensemble_group_tree_data.py

```diff
@@ -1,44 +1,45 @@
 import logging
-from typing import Any, Dict, List, Set, Tuple
+from typing import Any, Dict, List, Tuple
 
 import numpy as np
 import pandas as pd
 from webviz_config.common_cache import CACHE
 
+from webviz_subsurface._providers import EnsembleSummaryProvider
 
-class GroupTreeData:
-    """This class holds the summary and gruptree datasets and functionality
+
+class EnsembleGroupTreeData:
+    """This class holds the summary provider and gruptree dataset. It has functionality
     to combine the data and calculate the GroupTree component input dataset.
     """
 
-    def __init__(self, smry: pd.DataFrame, gruptree: pd.DataFrame):
-        self.smry = smry
+    def __init__(self, provider: EnsembleSummaryProvider, gruptree: pd.DataFrame):
+
+        self.provider = provider
         self.gruptree = gruptree
-        self.ensembles: Set[str] = gruptree["ENSEMBLE"].unique()
-        self.wells: Set[str] = gruptree[gruptree["KEYWORD"] == "WELSPECS"][
+
+        self.wells: List[str] = gruptree[gruptree["KEYWORD"] == "WELSPECS"][
             "CHILD"
         ].unique()
 
-        # Check that all field rate summary vectors exist
-        self.check_that_sumvecs_exists(["FOPR", "FGPR", "FWPR", "FWIR", "FGIR"])
-
-        # Check that WSTAT exists for all wells
-        self.check_that_sumvecs_exists([f"WSTAT:{well}" for well in self.wells])
+        # Check that all field rate and WSTAT summary vectors exist
+        self.check_that_sumvecs_exists(
+            ["FOPR", "FGPR", "FWPR", "FWIR", "FGIR"]
+            + [f"WSTAT:{well}" for well in self.wells]
+        )
 
-        # Check if the ensembles have waterinj and/or gasinj
-        self.has_waterinj, self.has_gasinj = {}, {}
-        for ensemble in self.ensembles:
-            smry_ens = self.smry[self.smry["ENSEMBLE"] == ensemble]
-            self.has_waterinj[ensemble] = smry_ens["FWIR"].sum() > 0
-            self.has_gasinj[ensemble] = smry_ens["FGIR"].sum() > 0
+        # Check if the ensemble has waterinj and/or gasinj
+        smry = self.provider.get_vectors_df(["FWIR", "FGIR"], None)
+        self.has_waterinj = smry["FWIR"].sum() > 0
+        self.has_gasinj = smry["FGIR"].sum() > 0
 
         # Add nodetypes IS_PROD, IS_INJ and IS_OTHER to gruptree
-        self.gruptree = add_nodetype(self.gruptree, self.smry)
+        self.gruptree = add_nodetype(self.gruptree, self.provider, self.wells)
 
         # Add edge label
         self.gruptree["EDGE_LABEL"] = self.gruptree.apply(get_edge_label, axis=1)
 
-        # Get summary data with metadata (ensemble, nodename, datatype, edge_or_node)
+        # Get summary data with metadata (nodename, datatype, edge_or_node)
         self.sumvecs: pd.DataFrame = self.get_sumvecs_with_metadata()
```

Comment:
Consider protected attribute notation for protected class attributes, i.e. underscore prefix: `self._provider`, `self._gruptree`, `self._sumvecs` etc.